unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarsLODController.cs

```diff
@@ -0,0 +1,147 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL
+{
+    public class AvatarsLODController : MonoBehaviour
+    {
+        public static AvatarsLODController i { get; private set; }
+
+        private const float LODS_LOCAL_Y_POS = 1.8f;
+        private const float LODS_VERTICAL_MOVEMENT = 0.1f;
+        private const float LODS_VERTICAL_MOVEMENT_DELAY = 1f;
+
+        public float lodDistance = 16f;
+        public int maxNonLODAvatars = 20;
+
+        private List<AvatarShape> avatarsList = new List<AvatarShape>();
+        private float lastLODsVerticalMovementTime = -1;
+
+        void Awake()
+        {
+            if (i != null)
+            {
+                Destroy(gameObject);
+                return;
+            }
+            i = this;
+
+            enabled = false;
+            KernelConfig.i.EnsureConfigInitialized()
+                        .Then(config =>
+                        {
+                            // We need this check because in the CI the KernelConfig tests may have this object destroyed somehow
+                            if (i == null || this == null)
+                                return;
+
+                            if (config.features.enableAvatarLODs)
+                            {
+                                enabled = true;
+                                CommonScriptableObjects.playerUnityPosition.OnChange += OnMainPlayerReposition;
+                            }
+                            else
+                            {
+                                enabled = false;
+                            }
+                        });
+        }
+
+        public void LateUpdate()
```

Comment:
Why we need it in the `LateUpdate`? Can't we use it in the normal `Update`?

Reply:
It's for the "Billboard" behaviour, if we use Update() instead of LateUpdate() there were issues in the past with the BillboardComponent not updating AFTER the main character moved, but in this case I think it's not needed, so I changed it to a regular `Update()` as you suggest.
If we change the controller to stop being a monobehaviour I'll remove it though.